src/tools/compiletest/src/runtest.rs

```diff
@@ -2042,10 +2050,24 @@ impl<'test> TestCx<'test> {
             if !is_rustdoc {
                 if let Some(ref linker) = self.config.linker {
                     rustc.arg(format!("-Clinker={}", linker));
+                } else {
+                    lld = self
+                        .config
+                        .target_rustcflags
+                        .as_ref()
+                        .map(|f| f.contains("-fuse-ld=lld"))
+                        .unwrap_or(false);
                 }
             }
         }
 
+        // Use a single thread for efficiency and a deterministic error message order
+        rustc.arg("-Zthreads=1");
+
+        if lld {
+            rustc.arg("-Clink-arg=-Wl,--threads=1");
```

Comment:
Not sure if this will work on all platforms. I onl the gcc + gold and gcc + lld combinations.

Reply:
`-Wl` won't work for any `-Clinker-flavor` other than `gcc`, and maybe a couple others. While this particular piece of code is not incorrect, because `-fuse-ld` is also a flag specific to `gcc`-like linkers, but its also true that this change is not inclusively supporting all the platforms where `linker-flavor` is not `gcc` by default.

Reply:
>  While this particular piece of code is not incorrect (because -fuse-ld is also a flag specific to gcc-like linkers)

Bootstrap configuration is not available in compiletest except for what gets mangled into `--target-rustcflags`. Should I add an explicit linker-type argument instead?

> -Wl won't work for any -Clinker-flavor other than gcc

Does rustc have any abstraction that handles linker args across compilers? I also tried `-Zpre-link-arg` but that also added it as a `cc` arg without compiler-specific mangling.


Or I could leave things as they are and only the gcc+ldd combo gets this particular performance optimization.

Reply:
I think its fine to leave as is.

There's no linker-flavor specific handling – in current rust setup a `gcc`-compatible tool itsself is itself considered a linker (without a consideration what it might invoke) and so the "linker arguments" are equivalent to gcc arguments in that mode.